lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -55,6 +56,213 @@ namespace Js
     }
 #endif
 
+
+    template<typename FIELDS, uint8 size, uint8 _MaxCount>
+    AuxPtrsFix<FIELDS, size, _MaxCount>::AuxPtrsFix()
+    {
+        this->count = AuxPtrsFix<FIELDS, 16>::MaxCount;
+        for (uint8 i = 0; i < count; i++)
+        {
+            this->type[i] = FIELDS::e_invalid;
+        }
+    }
+
+    template<typename FIELDS, uint8 size, uint8 _MaxCount>
+    AuxPtrsFix<FIELDS, size, _MaxCount>::AuxPtrsFix(AuxPtrsFix<FIELDS, 16>* ptr16)
+    {
+        this->count = AuxPtrsFix<FIELDS, 32>::MaxCount;
+        for (uint8 i = 0; i < AuxPtrsFix<FIELDS, 16>::MaxCount; i++)
+        {
+            this->type[i] = ptr16->type[i];
+            this->ptr[i] = ptr16->ptr[i];
+        }
+        for (uint8 i = AuxPtrsFix<FIELDS, 16>::MaxCount; i < count; i++)
+        {
+            this->type[i] = FIELDS::e_invalid;
+        }
+    }
+    template<typename FIELDS, uint8 size, uint8 _MaxCount>
+    inline void* AuxPtrsFix<FIELDS, size, _MaxCount>::get(FIELDS e)
+    {
+        if (_MaxCount == 1)
+        {
+            // specialize for x64 one pointer case
+            return e == type[0] ? ptr[0] : nullptr;
```

Comment:
only on x64, when initializing first pointer, it'll allocate 16 bytes, which can only hold 1 pointer (first 8 byte is metadata) which maxcount is 1. this line is specialize for x64 so it does not go into the loop and the compiler can optimize it trough inlining.